Backends/PDF/graphics.lisp

```diff
@@ -124,50 +138,67 @@ end-angle are NIL and filled is T, then a solid ellipse is drawn. If
 START-ANGLE and END-ANGLE are specified, draws an ellipse arc between
 the two angles, or a pie-wedge if filled is T."
   (let ((first-segment t))
-    (let ((theta (transform-angle tr theta)))
-      (flet ((draw-ellipse-segment (lambda1 lambda2)
-               (multiple-value-bind (p1x p1y q1x q1y q2x q2y p2x p2y)
-                   (ellipse-cubic-bezier-points (- lambda1 theta) (- lambda2 theta)
-                                                center-x center-y
-                                                a b theta)
-                 (with-transformed-position (tr p1x p1y)
-                   (with-transformed-position (tr q1x q1y)
-                     (with-transformed-position (tr q2x q2y)
-                       (with-transformed-position (tr p2x p2y)
-                         (when first-segment
-                           (pdf:move-to p1x p1y)
-                           (setf first-segment nil))
-                         (pdf:bezier-to q1x q1y q2x q2y p2x p2y))))))))
-        (if (or (and start-angle end-angle
-                     (zerop start-angle) (= (* pi 2) end-angle))
-                (and start-angle end-angle
-                     (zerop end-angle) (= (* pi 2) start-angle)))
-            (let* ((sweep (* pi 2))
-                   (steps 16)
-                   (step (/ sweep steps)))
-              (let ((seg-start-angle 0)
-                    (seg-end-angle (+ start-angle step)))
-                (loop for i below steps
-                   do (draw-ellipse-segment seg-start-angle seg-end-angle)
-                     (incf seg-start-angle step)
-                     (incf seg-end-angle step))))
-            (flet ((clamp-angle (angle)
-                     (min (max 0 angle) (* pi 2))))
-              (let ((start-angle (- (transform-angle tr (clamp-angle start-angle))))
-                    (end-angle (- (transform-angle tr (clamp-angle end-angle)))))
-                (let* ((sweep (- end-angle start-angle))
-                       (segment-count (ceiling (abs (/ sweep (/ pi 4)))))
+    (flet ((draw-ellipse-segment (lambda1 lambda2)
+             (multiple-value-bind (p1x p1y q1x q1y q2x q2y p2x p2y)
+                 (ellipse-cubic-bezier-points lambda1 lambda2
+                                              center-x center-y
+                                              a b theta)
+               (with-transformed-position (tr p1x p1y)
+                 (with-transformed-position (tr q1x q1y)
+                   (with-transformed-position (tr q2x q2y)
+                     (with-transformed-position (tr p2x p2y)
```

Comment:
maybe `with-transformed-positions` macro will be more elegant here?

Reply:
At the expense of consing up a coord-seq? I don't think that's worth it.